tests/FakeItEasy.Specs/UserCallbackExceptionSpecs.cs

```diff
@@ -260,6 +262,30 @@ public void ExceptionInReturnValueProducer(IFoo fake, Exception exception)
                 .x(() => exception.InnerException.Should().BeAnExceptionOfType<MyException>().Which.Message.Should().Be("Oops"));
         }
 
+        [Scenario]
+        public void ExceptionInOutAndRefValueProducer(IFoo fake, Exception exception)
+        {
+            "Given a fake"
+                .x(() => fake = A.Fake<IFoo>());
+
+            int x = 0;
+            string s;

```

Comment:
Not a huge deal, but it's surprising to see local variables. These don't work as parameters? Or you prefer as locals? Either answer is fine!

Reply:
Not sure why I did this. I guess I always found it strange to use parameters for things that are conceptually local variables. Or maybe there was a technical reason. I'll have a look.

Reply:
Moved the local variables to parameters.

Reply:
Thanks. I'm of the same mind as you: it's a little strange to use parameters for local variables. On the other hand, after doing it for so long in the code, if felt strange _not_ to!